_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+
+Clone the modified JHipster reactive microservice project from GitHub and checkout the `start` tag.
+
+```bash
+git clone git@github.com:moksamedia/okta-azure-kubernetes-cosmodb.git
+cd okta-azure-kubernetes-cosmodb
+git fetch --all --tags
+git checkout tags/start -b working
+```
+
+#### Create the Azure Cosmos DB Mongo Database
+
+You need to create an Azure Cosmos DB instance. You can either use the [Azure Portal](portal.azure.com) or the CLI to create a new Cosmos DB instance. Make sure you create a one that is **Azure Cosmos DB API for MongoDB** (Cosmos DB supports various database types). If you use the portal, it's pretty self-explanatory but don't forget to enable the free tier and enable a public network.
+
+Here are the instructions for using the CLI. This assumes you have already logged into the Azure CLI (if not, do so first using `az login` from a Bash shell).
+
+Open a Bash shell. Create a resource group with the following command.
+
+```bash
+az group create --name australia-east --location australiaeast
+```
+
+Create the Cosmos DB account in the resource group. Substitute your Azure subscription name in the command below (it's probably `Azure subscription 1`, which is what mine defaulted to).
+
+```bash
+az cosmosdb create --name jhipster-cosmosdb --resource-group australia-east --kind MongoDB --subscription <you-subscription-name> --enable-free-tier true --enable-public-network true
+```
+
+Once that command returns (it may take a few minutes), it should list a lot of JSON showing properties of the created Cosmos DB account.
+
+If you get an error that says`(BadRequest) DNS record for cosmosdb under zone Document is already taken.` you need to change the `--name` parameter to something else.
+
+I'm using the Australia East location because that was the location that had free tier AKS nodes available when I wrote this tutorial. You can use any resource group you want as long as it allows you to create the AKS cluster later in the tutorial. Even if you can't use the free tier or the free credits, if you stop and start the AKS cluster between working on the tutorial, the cost should be very small (mine was less than a few dollars). The application should still work if the Cosmos DB database is in a different resource group and region since the database URI is configured to be publicly accessible.
+
+List the connection string for the Cosmos DB API for MongoDB endpoint using the following command.
+
+```bash
+az cosmosdb keys list --type connection-strings --name jhipster-cosmosdb --resource-group australia-east
+```
+
+This will list four connection strings. You need to save (copy and paste somewhere) the first, the primary connection string. (Ellipses have been used for brevity below)
+
+```bash
+  "connectionStrings": [
+    {
+      "connectionString": "mongodb://jhipster-cosmosdb:XBq5KZ81V8hM63KjCOezi1arq...,
+      "description": "Primary MongoDB Connection String"
+    },
+    ...
+  ]
+}
```

Comment:
Curly braces don't match up. Should there be something like this at the beginning of this code block?

```
{
  ...
```